From 14c9714ba83e8246f61e72275ce22f8b43d70e4c Mon Sep 17 00:00:00 2001
From: Xianzhu Wang <wangxianzhu@chromium.org>
Date: Mon, 14 Oct 2024 22:25:22 +0000
Subject: [PATCH] Speculative fix of crash in ScrollTree::SetScrollOffset()

Now tolerate null GetSyncedScrollOffset(id).

crrev.com/c/5823800 fixed crashes at the same location for overflow:hidden but didn't fix all of the crashes. The remaining
crashes may be in a similar situation: a blink scroll node is not
in LocalFrameView::UserScrollableAreas(), is not painted, doesn't
force main-thread scrolling, and has a scroll animation. It looks
safe to ignore the null GetSyncedScrollOffset in the situation.

Fixed: 371968395
Change-Id: I58cee0eb79e53102fde9b0af3691bc88be440198
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5931266
Reviewed-by: Philip Rogers <pdr@chromium.org>
Commit-Queue: Xianzhu Wang <wangxianzhu@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1368456}
---

--- a/cc/trees/property_tree.cc
+++ b/cc/trees/property_tree.cc
@@ -1940,8 +1940,9 @@
   }
 
   if (property_trees()->is_active()) {
-    DCHECK(GetSyncedScrollOffset(id));
-    return GetSyncedScrollOffset(id)->SetCurrent(scroll_offset);
+    if (auto* synced_scroll_offset = GetSyncedScrollOffset(id)) {
+      return synced_scroll_offset->SetCurrent(scroll_offset);
+    }
   }
 
   return false;
